### PR TITLE
BItcoin and Electrum servers

### PR DESCRIPTION
**Pull request by abbradar**
###### Motivation for this change

Add services to keep your own Electrum server node.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Notice that while that seems to work, I haven't been able to use it because Electrum server is _very_ inefficient, to the point that my server is not fast enough to be synced with the network.

### CODE REVIEW

File: nixos/modules/services/misc/bitcoind.nix [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    services.bitcoind = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable Bitcoin daemon";
+      };
+
+      dataDir = mkOption {
+        type = types.str;
```
Comment (joachifm) [inline nixos/modules/services/misc/bitcoind.nix]: Maybe use `types.path`?

Comment (abbradar) [inline nixos/modules/services/misc/bitcoind.nix]: Good call! Updated.

**Comment by koenigmaximilian** [comment]
I'm gonna publish my bitcoin-core, dashcore, parity and monero modules today or tomorrow.

**Comment by abbradar** [comment]
Doesn't seem to spark any interest and I don't use those servers myself; closing.

**Pull request closed without merging**